File: examples/num_devices.rs
use libftd2xx_ffi::{DWORD, FT_LIST_NUMBER_ONLY, FT_ListDevices, FT_STATUS, PVOID};

fn main() {
    let mut num_devs: DWORD = 5;
    let dummy: PVOID = std::ptr::null_mut();
    let status: FT_STATUS = unsafe {
        FT_ListDevices(
            (&raw mut num_devs).cast::<std::ffi::c_void>(),
            dummy,
            FT_LIST_NUMBER_ONLY,
        )
    };
    println!("Status: {status}");
    println!("Number of devices: {num_devs}");
}

File: examples/num_devices.c
#include <stdio.h>
#include "../libftd2xx_src/linux/x64/ftd2xx.h"

/*
* This C program is equivalent to the `num_devices` rust example.
* This is useful to compare the native C implementation with the rust bindings.
*
* Compile with:
* clang num_devices.c -lftd2xx -L../libftd2xx_src/linux/x64/build -lpthread
*/
int main(void) {
    DWORD num_devs;
    FT_STATUS status = FT_ListDevices(&num_devs, NULL, FT_LIST_NUMBER_ONLY);
    printf("Status: %d\n", status);
    printf("Number of devices: %d\n", num_devs);
    return 0;
}